.github/workflows/benchmark.yml

```
@@ -38,6 +44,28 @@ jobs:
           key: ${{ runner.os }}-${{ github.sha }}
           restore-keys: ${{ runner.os }}
 
+      - name: Cache test data directory
+        id: cache-test-data
+        uses: actions/cache@v2
+        with:
+          path: |
+            ${{ env.IRIS_TEST_DATA_PATH }}
+          key:
+            test-data-${{ env.IRIS_TEST_DATA_VERSION }}-${{ env.TEST_DATA_CACHE_BUILD }}
+
+      - name: Fetch the test data
+        if: steps.cache-test-data.outputs.cache-hit != 'true'
+        run: |
+          wget --quiet https://github.com/SciTools/iris-test-data/archive/v${IRIS_TEST_DATA_VERSION}.zip -O iris-test-data.zip
+          unzip -q iris-test-data.zip
+          mkdir --parents ${GITHUB_WORKSPACE}/${IRIS_TEST_DATA_PATH}
+          mv iris-test-data-${IRIS_TEST_DATA_VERSION} iris-test-data
+          mv iris-test-data ${GITHUB_WORKSPACE}/benchmarks
+          
+      - name: Set test data var
+        run: |
+          echo "OVERRIDE_TEST_DATA_REPOSITORY=${GITHUB_WORKSPACE}/${IRIS_TEST_DATA_PATH}/test_data" >> $GITHUB_ENV
+
       - name: Run CI benchmarks
         run: |
           mkdir --parents benchmarks/.asv
```

During reviewing of this PR I've discovered that the below piping hides the exit status so that benchmarking failures are not detected!

```suggestion
          mkdir --parents benchmarks/.asv
          set -o pipefail
```